### DISCUSSION TITLE
GitHub markdown, alerts are not working when they have been indented

### DISCUSSION THREAD

**Discussion by kasir-barati** [community/community#118296] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hi dear community,

I am fond of how beautiful are alerts in GitHub markdowns and I am using them whenever needed but I realized today that if I indent them GitHub cannot render them. Just look at this repo: https://github.com/kasir-barati/you-say

In [this section](https://github.com/kasir-barati/you-say?tab=readme-ov-file#github-script-action) I am indenting the "caution" to show that it is part of that list but to no avail, it is not working [the way it does when I am not indenting it](https://github.com/kasir-barati/you-say?tab=readme-ov-file#workflows)

Here I attach two screen shots: 
Working case -- without indentation:
![image](https://github.com/community/community/assets/73785723/2f8cc44a-e2f8-452d-8a6e-7981b2916773)
Broken case -- with indentation:
![image](https://github.com/community/community/assets/73785723/72b75173-2283-412a-8461-fc874c6afe6d)

Are we restricted/limited to use them just without indentation? if yes where can I open an issue for them so we can discuss it further to see if it is viable and logically speaking practical to be added.

Many thanks

**Comment by jin-qin** [comment]
I am having the same issue, it should be a bug as I don't have this issue when I am previewing the same file in VS Code.

It's been three months since this question was created, could someone from Github help here?

**Comment by jcote-tc** [comment]
I'm having the same issue. i.e.

This works when the Alert is at the beginning of line:

1. Title of first section

   This is the first section.

> [!NOTE]
> This is a note that creates the Alert icon correctly.

This doesn't work, when I try to align with another paragraph by adding three spaces in front of the line:

1. Title of first section

   This is the first section.

   > [!NOTE]
   > This is a note that creates a quoted text. It doesn't create the Alert icon.

**Comment by JaimeStill** [comment]
The documentation for [alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) does state:

> Alerts cannot be nested within other elements.

I agree with @kasir-barati that this restriction should be considered further because it severely limits the utility of alerts.

**Comment by jjmaestro** [comment]
+1 to having Alerts work in nested elements. At the very least, I'd love if they would work inside [collapsed sections](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections) because that really helps to have easy-to-read `README` files and losing the ability to highlight stuff with alerts is a big loss / tradeoff VS having to do without collapsed sections.

**Reply by frozenfrank** [reply]
@jjmaestro Great comment!

I discovered this issue independently and have created a discussion specifically about alerts within collapsed sections. See #143379.

**Comment by Manuchexra** [comment]
ok

**Reply by Manuchexra** [reply]
> ok

**Comment by bstrdsmkr** [comment]
:+1: ing and adding my use case: a Note admonition associated with an item in a bulleted list. Example:
```
1. Here's a thing
    ```
    with.some(code)
    ```
    > [!NOTE] 
    > The variable code is defined in foo.bar
1. Next, frogblast the vent cores
```
But the admonition renders as a block quote due to 4 spaces required to keep the bulleted list going:
---
1. Here's a thing
    ```
    with.some(code)
    ```
    > [!NOTE] 
    > The variable code is defined in foo.bar
1. Next, frogblast the vent cores
---
> [!NOTE] 
> The the literal `1.` above render correctly as `1.` and `2.` per markdown syntax

You can elide the four spaces and the admonition will render correctly, but the bulleted list will not continue:
---
1. Here's a thing
    ```
    with.some(code)
    ```
> [!NOTE] 
> The variable code is defined in foo.bar
1. Next, frogblast the vent cores
---
> [!CAUTION] 
> The the literal `1.`'s both render as `1.` instead of the expected incremented item count

You can manually change the numbers, but of course that relies on users noticing and correcting the problem, which in practice doesn't happen as often as you'd like

**Comment by JosephMarinier** [comment]
Another use case is an alert inside a `<details>` block:

## Example
```Markdown
<details>
<summary>Click to expand</summary>

Other _Markdown_ elements work.

> [!NOTE]
> But an alert does not work since it is "nested", even though it is not indented.

</details>
```

## Result
<details>
<summary>Click to expand</summary>

Other _Markdown_ elements work.

> [!NOTE]
> But an alert does not work since it is "nested", even though it is not indented.

</details>

**Comment by klaudiagrz** [comment]
I'd like to upvote this issue – I also found all the use cases mentioned above quite problematic, since alert panels nested within a bulleted list are not a rare use case. It would be super cool if you could extend it 🙏🏻

**Comment by DanielBUBU** [comment]
Sadly this is still unanswered

**Comment by udlose** [comment]
I'd love to see GitHub reconsider the current state of not allowing Alerts to be nested inside of other elements. Having to keep Alerts (which I love) outside of collapsible content makes for very ugly and inconsistent-looking documentation. 

Please consider adding the ability to nest Alerts inside of other elements!

**Comment by JoKeyser** [comment]
Indented Alerts work fine on GitLab, would be nice if GitHub can fix this problem.

**Comment by joshenders** [comment]
Like many others before me, I'm here to express my support of this issue.
